### PR TITLE
Grant Type: Client credentials

### PR DESCRIPTION
**Pull request by jdelaune**
Hey,

Started by implementing the client credentials grant type. Hopefully the way I've implemented it makes sense. Tried to keep it open and as modifiable as possible by other bundles. Also using Doctrine2 Entities for the storage.

Let me know what you think and I can start to do the rest.

### CODE REVIEW

File: Controller/TokenController.php [hunk 1]
```diff
@@ -19,13 +17,9 @@ class TokenController extends Controller
      */
     public function tokenAction()
     {
-        // get the oauth server (@see OAuth2/ServerBundle/Resources/config/services.xml)
-        $server = $this->get('oauth2.server');
-
-        // get the oauth response object (@see OAuth2/ServerBundle/Resources/config/services.xml)
-        $response = $this->get('oauth2.response');
+        $server = $this->get('c4.oauth2.server');
+        $server->addGrantType($this->get('c4.oauth2.grant_type.client_credentials'));
```
Comment (bshaffer) [inline Controller/TokenController.php]: why the c4 prefix?

File: DependencyInjection/OAuth2ServerExtension.php [hunk 2]
```diff
@@ -11,12 +11,10 @@ class OAuth2ServerExtension extends Extension
...
     {
-        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
-        $loader->load('services.xml');
-    }
+        $configuration = new Configuration();
+        $config = $this->processConfiguration($configuration, $configs);
 
-    public function getAlias()
-    {
-        return 'o_auth2_server';
```
Comment (bshaffer) [inline DependencyInjection/OAuth2ServerExtension.php:20]: why did you remove the `getAlias` method?

File: Resources/config/services.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        arguments:
+            em: @doctrine.orm.entity_manager
+    oauth2.storage.access_token:
+        class: %oauth2.storage.access_token.class%
+        arguments:
+            em: @doctrine.orm.entity_manager
+    oauth2.grant_type.client_credentials:
+        class: %oauth2.grant_type.client_credentials.class%
+        arguments:
+            storage: @oauth2.storage.client_credentials
```
Comment (bshaffer) [inline Resources/config/services.yml]: I don't mind yml, but the bundles I've seen seem to implement `xml` as the standard

**Comment by bshaffer** [comment]
This is awesome!  I had a few questions, but otherwise this looks great

**Comment by jdelaune** [comment]
Ah yes sorry was converting a bundle I had previously made as I thought it would be better to work on this than a separate project. I think I've fixed things up and using lovely un-readable XML instead of YAML :P

**Comment by bshaffer** [comment]
Lol yeah, thanks.  I actually agree with you, but I conform to standards.

Thank you so much!  This is a great start.
